presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestTransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.iterative.rule.test.BaseRuleTest;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.facebook.presto.tpch.TpchColumnHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.any;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestTransformCorrelatedSingleRowSubqueryToProject
+        extends BaseRuleTest
+{
+    @Test
+    public void testDoesNotFire()
+    {
+        tester().assertThat(new TransformCorrelatedSingleRowSubqueryToProject())
+                .on(p -> p.values(p.symbol("a")))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testRewrite()
+    {
+        tester().assertThat(new TransformCorrelatedSingleRowSubqueryToProject())
+                .on(p ->
+                        p.lateral(
+                                ImmutableList.of(p.symbol("nationkey"), p.symbol("nationkey")),
+                                p.tableScan(ImmutableList.of(p.symbol("nationkey")),
+                                        ImmutableMap.of(p.symbol("nationkey"), new TpchColumnHandle("tpch:nationkey",
+                                                BIGINT))),
+                                p.project(
+                                        Assignments.of(p.symbol("nationkey"), expression("nationkey")),
```

Comment:
this is identity projection, can you use something more complicated like `a +1`?

Reply:
There is no need to use `Ack`. Notice that you sent emails with these messages. We typically use `reactions` for that (see your message were I used one). Then you could use `thumbs up` and I will know that you agree with the comment. Reactions do not send emails.

We receive a lot of emails, I hope that getting one less will make our work a bit more efficient ;).